contracts/QuestFactory.sol

```diff
@@ -45,7 +51,9 @@ contract QuestFactory is Initializable, OwnableUpgradeable {
         string memory contractType_,
         string memory questId_,
         address receiptContractAddress_
-    ) public onlyOwner returns (address) {
+    ) public returns (address) {
+        if (!hasRole(CREATE_QUEST_ROLE, msg.sender)) revert InavlidRoleToCreateQuest();
```

Comment:
Any benefit for using this over using OZ's `onlyRole` modifier?

Reply:
no not really, the only reason was for the `InavlidRoleToCreateQuest` custom error, but I'm happy to move it to the modifier too, less code that way.